crates/storage/storage.rs

```diff
@@ -270,12 +270,20 @@ impl Store {
     }
 
     pub fn add_initial_state(&mut self, genesis: Genesis) -> Result<(), StoreError> {
-        // TODO: Check initial state is not already present in db
         info!("Storing initial state from genesis");
 
         // Obtain genesis block
         let genesis_block = genesis.get_block();
 
+        if let Some(header) = self.get_block_header(genesis_block.header.number)? {
+            if header.compute_block_hash() == genesis_block.header.compute_block_hash() {
+                info!("Found this initial state on storage, nothing to do");
```

Comment:
I'm not sure what extra data would be relevant, considering the previous `info` also says little. I added this mostly so the user knows it did nothing.

Reply:
We should be able to understand what is happening when we see this on the logs, with this message we don't know what the "initial_state" refers too. Maybe it can be more specific like "Received genesis file matching the one stored, nothing to do". So that we know this refers to genesis and not some other stored state

Reply:
The previous line would be `"Storing initial state from genesis"`, from the other `info!` line. But I can rephrase to make it more self-contained.